tutorials/fiori-ios-scpms-starter-mission-03/fiori-ios-scpms-starter-mission-03.md

```diff
@@ -153,6 +166,40 @@ The Product List is a Table View Controller which means the structure is similar
 
     ```
 
+    **For Offline OData**
+
+    Add the following import statement to your class:
+
+    ```Swift
+    import SAPOfflineOData
+    import SAPOData
+
+    ```
+
+    Implement the following lines of code directly below the logger instance as class properties:
+
+    ```Swift
+
+    /// First retrieve the destinations your app can talk to from the AppParameters.
+    let destinations = FileConfigurationProvider("AppParameters").provideConfiguration().configuration["Destinations"] as! NSDictionary
+
+    var dataService: ESPMContainer<OfflineODataProvider>? {
+        guard let odataController = OnboardingSessionManager.shared.onboardingSession?.odataControllers[destinations["com.sap.edm.sampleservice.v2"] as! String] as? Comsapedmsampleservicev2OfflineODataController, let dataService = odataController.espmContainer else {
+            AlertHelper.displayAlert(with: NSLocalizedString("OData service is not reachable, please onboard again.", comment: ""), error: nil, viewController: self)
+            return nil
+        }
+        return dataService
+    }
+
+    private let appDelegate = UIApplication.shared.delegate as! AppDelegate
+    private let logger = Logger.shared(named: "ProductsTableViewController")
+
+    private var imageCache = [String: UIImage]()
+    private var productImageURLs = [String]()
+    private var products = [Product]()
+
+    ```
+
     You might wonder why we are not passing the data service in from the Overview. Later on when adapting the app to work on MacOS through Mac Catalyst, the user will have the option to jump directly into the product list. In that case you won't perform a segue and so on not be able to pass in the data service. Of course there are ways to refactor this into a more centralized way but for simplicity reason we stick to this approach.
```

Comment:
Please don't say "we", say "you" instead as described in the [style guide](https://jam4.sapjam.com/wiki/show/MozDWaPczznL8NKeGEk0y9?_lightbox=true#mcetoc_1dtqgno1sd).